### ISSUE TITLE
Multiple IDPs/Roles/Profiles

### ISSUE THREAD

**Issue by andrew-kline** [okta/okta-aws-cli#36]
Hi friends 👋 

One of my favorite aspects of gimme-aws-creds is the ability to authenticate to multiple accounts+roles+profiles at the same time. Writing a shell script with all of the IDP/role/profile combinations that I need is fine, but having this configurable natively seems like a feature that others might benefit from as well.

Thanks in advance!
Andrew

**Comment by philhead** [comment]
I agree. Having to run okta-aws-cli each time I need to assume a different role is painful, and this behaviour will break some of our existing scripts and Terraform. The tool should write all of the user's assigned IAM roles (from all configured IdPs) to separate profiles in .aws/credentials (as does gimme-aws-creds).

**Comment by monde** [comment]
@andrew-kline @philhead please open support tickets https://support.okta.com/ to request these features, this is something for the project manager to consider. Our GA goal was to provide real OIE support for AWS Federated apps and was not to be a direct replacement for Nike Gimme creds. FWIW, even with token expiry time limits, writing to a credentials file is a security anti-pattern and we didn't really want to support this practice.

**Comment by philhead** [comment]
Support case opened: https://ideas.okta.com/app/#/case/169841

**Comment by monde** [comment]
cc: @jefftaylor-okta

**Comment by niradler** [comment]
any update here? should we open another ticket ?

**Comment by dcgoodridge** [comment]
Tried #94, building windows-amd64 and linux-amd64 bins and it works as a charm ✅ . Thanks @daniel-sampliner

**Comment by monde** [comment]
Great to hear it illustrates behavior some users desire. However, the there isn't info in the README how the feature works, for instance, assumptions about IAM Role permissions, and how first run is essentially a config run. Also, the feature isn't gated by a CLI flag assuming everyone wants to use this functionality. I'm considering the PR a proof of concept. @jefftaylor-okta and have plans on this feature as well caching access tokens. I'm on holiday fir the week and we are looking for a v1.0.0 release before 4/30/23, stay tuned.

**Comment by philhead** [comment]
@monde Is there a new target date for this feature?

**Comment by monde** [comment]
I don't have an update but am ack'ing we are talking about this currently.

**Comment by monde** [comment]
Okta internal reference: https://oktainc.atlassian.net/browse/OKTA-641192
